IPNS.md

```diff
@@ -91,24 +178,59 @@ As soon as the node has the most recent record, the signature and the validity m
 
 Finally, the network nodes may also republish their records, so that the records in the network continue to be valid to the other nodes.
 
-## Overview
-
-![](img/ipns-overview.png)
-
-## API Spec
-
-  - [API_CORE](https://github.com/ipfs/specs/blob/master/API_CORE.md)
+### Record Creation
+
+IPNS record MUST be serialized as `IpnsEntry` protobuf and the raw record data MUST be signed using the private key.
+
+Creating a new IPNS record MUST follow the below steps:
+
+1. Create `IpnsEntry` and set `value`, `validity`, `validityType`, `sequence`, and `ttl`
+   - If you are updating an existing record, remember to increase values in `sequence` and `validity`
+2. Create a DAG-CBOR document with the same values for `value`, `validity`, `validityType`, `sequence`, and `ttl`
+3. Store DAG-CBOR in `IpnsEntry.data`.
+   - If you want to store additional metadata in the record, add it under unique keys at `IpnsEntry.data`.
+   - The order of fields impacts signature verification. If you are using an alternative CBOR implementation, make sure the CBOR field order follows [RFC7049](https://www.rfc-editor.org/rfc/rfc7049) sorting rules: length and then bytewise. The order of fields impacts signature verification.
+4. If your public key can't be inlined inside the IPNS Name, include a serialized copy in `IpnsEntry.pubKey`
+   - This step SHOULD be skipped for Ed25519 keys.
+6. Create bytes for signing by concatenating `ipns-signature:` prefix (bytes in hex: `69706e732d7369676e61747572653a`) with raw CBOR bytes from `IpnsEntry.data`
+7. Sign bytes from the previous step using the private key, and store the signature in `IpnsEntry.signatureV2`
+
+### Record Validation
+
+Implementations MUST resolve IPNS Names using only verified records.
+Record's data and signature verification MUST be implemented as outlined below, and fail on the first error.
+
+1. Before parsing protobuf, confirm `IpnsEntry` size is less than 2 MiB
```

Comment:
Maybe "Before parsing the protobuf, confirm that `IpnsEntry` is less than 2MiB in size" would be better wording?

Reply:
```suggestion
1. Before parsing the protobuf, confirm that `IpnsEntry` is less than 2MiB in size
```